.github/workflows/generate_graph_test.yml

```diff
@@ -0,0 +1,81 @@
+        description: 'Repo to build kiota from'
+        type: string
+        default: 'https://github.com/microsoft/kiota.git'
+      OPEN_API_DEF:
+        description: 'OpenAPI definition to build test SDK for'
+        type: string
+        default: 'https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/openapi/v1.0/openapi.yaml'
+
+  push:
+    branches: [ nikithauc/model-interface-serialization ]
+  pull_request:
+
+env:
+  KIOTA_BRANCH: 'nikithauc/model-as-interfaces'
+  KIOTA_REPO: 'https://github.com/microsoft/kiota.git'
+  OPEN_API_DEF: 'https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/openapi/v1.0/openapi.yaml'
+
+jobs:
+    build:
+        runs-on: ubuntu-latest
+        environment:
+          name: build_test
+        env:
+          TENANT_ID: ${{ secrets.TENANT_ID }}
+        strategy:
+          matrix:
+            node-version: [14.x, 16.x, 18.x]
+        steps:
+        # download kiota
+        - run: git clone --branch ${{ github.event.inputs.KIOTA_BRANCH }} --single-branch ${{ github.event.inputs.KIOTA_REPO }} kiota_tool
+          if: ${{ inputs.KIOTA_BRANCH }}
+
+        - run: git clone --branch ${{ env.KIOTA_BRANCH }} --single-branch ${{ env.KIOTA_REPO }} kiota_tool
+          if: ${{ !inputs.KIOTA_BRANCH }}
+
+        # build kiota
+        - name: Setup .NET
+          uses: actions/setup-dotnet@v3
+          with:
+            dotnet-version: 7.0.x
+        - name: Build Kiota tool for linux-x64
+          run: dotnet publish ./kiota_tool/src/kiota/kiota.csproj -c Release -p:PublishSingleFile=true -r linux-x64 -o ./kiota-linux-x64 --self-contained
+        - run: chmod +x ./kiota-linux-x64/kiota
+        # use kiota to generate an SDK into the test folder
+        - run: ./kiota-linux-x64/kiota generate --openapi ${{ github.event.inputs.OPEN_API_DEF }} -l typescript --output ./packages/test/generatedCode --clean-output
+          if: ${{ inputs.KIOTA_BRANCH }}
+        - run: ./kiota-linux-x64/kiota generate --openapi ${{ env.OPEN_API_DEF }} -l typescript --output ./packages/test/generatedCode --clean-output
+          if: ${{ !inputs.KIOTA_BRANCH }}
+        - uses: actions/checkout@v3
+        - name: Use Node.js ${{ matrix.node-version }}
+          uses: actions/setup-node@v3
+          with:
+            node-version: ${{ matrix.node-version }}
+        - run: yarn install --frozen-lockfile --ignore-engines #spfx still not compatible with node 18
+        - run: yarn global add mocha@10.X
+        - name: Archive dist folders # archive dist folders to verify if they are transpiled correctly and available for publishing
+          uses: actions/upload-artifact@v3
+          with:
+            name: dist folders
+            path: |
+              packages/abstractions/dist
+              packages/serialization/json/dist
+              packages/serialization/text/dist
+              packages/http/fetch/dist
+              packages/authentication/azure/dist
```

Comment:
do we care about those dists as we're not going to publish from this workflow? Can we instead upload a webpack bundle analyzer report to have an history of the changes?

Reply:
That is the plan, I'm iterating on getting this working and then adding the functionality I want.

Reply:
So, because we don't build the SDK with webpack I don't think that we can use the webpack bundle analyzer for that aspect, but I've used `npm pack --dry-run --json` instead to get some package size data and output it as a PR comment.

My thinking is that I'll also add a small sample application which is built using webpack that uses the generated SDK and that way we can also track how we're doing on the bundle size for a client app and that would use WebPack.

Reply:
Marking this as resolved now that there is a sample app which we can report on the bundle size with.